### PR TITLE
Attaching volume to OpenTofu runner to read https certs

### PR DESCRIPTION
**Pull request by sudhir-kelkar**
This PR will mount cert path to opentofu-runner worker deployment by reading certs from secret `application-certificate-secret`

### CODE REVIEW

**Comment by Fryguy** [comment]
@bdunne Please review.

File: app/models/opentofu_worker.rb [hunk 1]
```diff
@@ -49,6 +49,13 @@ def unit_environment_variables
     }
   end
 
+  def configure_service_worker_deployment(definition)
+    super
+
+    definition[:spec][:template][:spec][:containers].first[:volumeMounts] << {:name => "cert-path", :mountPath => "/root"}
```
Comment (bdunne) [inline app/models/opentofu_worker.rb:55]: Does the container have access to `/root`?

File: app/models/opentofu_worker.rb [hunk 2]
```diff
@@ -49,6 +49,13 @@ def unit_environment_variables
...
+    definition[:spec][:template][:spec][:volumes] << {:name => "cert-path", :secret => {:secretName => "application-certificate-secret", :items => [{:key => "httpd_crt", :path => "server.crt"}, {:key => "httpd_key", :path => "server.key"}], :defaultMode => 420}}
```
Comment (bdunne) [inline app/models/opentofu_worker.rb:56]: `application-certificate-secret` should not be hard-coded.

Comment (bdunne) [reply app/models/opentofu_worker.rb:56]: `{:key => "httpd_crt"` This should have its own certificate, not reuse the one for httpd.

File: app/models/opentofu_worker.rb (hunk 1 above)
Comment (sudhir-kelkar) [reply app/models/opentofu_worker.rb]: Using this path now which is accessible from terraform runner container `/opt/app-root/src/`
Writing certificate and key on path `/opt/app-root/src/config/cert`

File: app/models/opentofu_worker.rb (hunk 2 above)
Comment (sudhir-kelkar) [reply app/models/opentofu_worker.rb]: Updated IM installer code to have new key and cert in secret `application-certificate-secret`.
Using those secret key for creating new volume mounts.
ref: https://github.ibm.com/IBMPrivateCloud/ibm-infra-management-install-operator/pull/566/files

Comment (sudhir-kelkar) [reply app/models/opentofu_worker.rb]: There is no environment variable I can use for reading secret name as `application-certificate-secret`. 
May be I am missing existing so please guide. Secondly If I define the variable and assign value to it as `application-certificate-secret`  it is as good as hard coding.

Comment (sudhir-kelkar) [reply app/models/opentofu_worker.rb]: Using environment variable API_SSL_SECRET_NAME instead of actual secret name

**Pull request merged**